rfcs/0064-documentation-format.md

```diff
@@ -0,0 +1,173 @@
+---
+feature: documentation-format
+start-date: 2019-12-31
+author: Silvan Mosberger (infinisil)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+The Nix community wants to move away from using Docbook as our documentation format and replace it with something else. However it is unclear what it should be replaced with. This RFC gives a concrete process for determining the new documentation format. It does NOT say what format should be used.
+
+# Motivation
+[motivation]: #motivation
+
+The current format for documentation of NixOS projects is DocBook. The format has been a discussion for many years for multiple reasons.
+This RFC describes a method for deciding on what format to use, and should allow us to decide on a format for the coming years, and improve our documentation situation.
+
+# Detailed design
+[design]: #detailed-design
+
+The process for determining the doc format is as follows:
+- A set of requirements for the doc format is decided through the RFC discussion
+- Doc format candidates are collected and evaluated to see if they fulfil the requirements.
+- A short objective overview of each valid candidate format is written, along with their advantages/disadvantages
+- The RFC is accepted
+- A [Discourse](https://discourse.nixos.org/) post is created with these overviews, along with a poll such that people can vote on the formats they prefer. This poll will be open to the whole community and should be advertised as such
+- Whatever format wins in the poll is chosen as the new default documentation format. If later it is discovered that the winner is infeasible for any reason, e.g. if it doesn't meet the requirements after all, the format on second place is chosen instead, and so on.
+
+## Poll
```

Comment:
I don’t like the poll idea. It’s too easy to vote for your favourite documentation format without considering the implications for our use case in Nixpkgs, which the most popular format might not be suited for.

It’s difficult to reasonably argue this without mentioning specific formats — I think it’s unreasonable to believe that formats and decision-making processes are unrelated. I think we all know which format is going to win a popularity contest, but that doesn’t mean it’s the best choice for Nixpkgs.

Reply:
Summarizing from the discussion we had on IRC:
- We need to make sure Markdown (and all others) can actually work as a format first with a small demo showing all the required features. Only if that is successful it can be a candidate. Sounds like a good idea to have small demos *before* the poll
- This might need some extensions to "standard" markdown, which themselves should be as standard as possible. We ideally shouldn't have our own "Nix flavored Markdown". E.g. Sphinx provides a bunch of extensions by default already
- If Markdown ends up in the poll, it might very well be a popularity contest, but that's not a bad thing, because everybody who knows markdown already won't have to learn many new things to contribute docs. We might have to educate people about how to use the extensions though